go/vt/vttablet/tabletmanager/replmanager.go

```diff
@@ -110,7 +132,10 @@ func (rm *replManager) checkActionLocked() {
 		// If only one of the threads is stopped, it's probably
 		// intentional. So, we don't repair replication.
 		if status.SQLThreadRunning || status.IOThreadRunning {
-			return
+			// Check if the Primary is still the source though
+			if primaryIsSource, err := rm.checkPrimaryIsSource(status); err == nil && primaryIsSource {
+				return
+			}
```

Comment:
In case we get an error here, the error will be from a topo call either for `GetShard` or for `GetTablet`. In either case, I don't think we should try to fix the replication since the repairReplication will try to access the topo too. We should return in that case and wait for the next tick like we do in case of errors in above lines.